src/libraries/System.Private.CoreLib/src/System/Collections/BitArray.cs

```diff
@@ -0,0 +1,972 @@
+        /// <exception cref="ArgumentException"><paramref name="value"/> and the current <see cref="BitArray"/> do not have the same number of elements.</exception>
+        public BitArray And(BitArray value) => Invoke<AndBinaryOp>(value);
+
+        /// <summary>
+        /// Performs the bitwise OR operation between the elements of the current <see cref="BitArray"/> object and the
+        /// corresponding elements in the specified array. The current <see cref="BitArray"/> object will be modified to
+        /// store the result of the bitwise OR operation.
+        /// </summary>
+        /// <param name="value">The array with which to perform the bitwise OR operation.</param>
+        /// <returns>An array containing the result of the bitwise OR operation, which is a reference to the current <see cref="BitArray"/> object.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="value"/> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="value"/> and the current <see cref="BitArray"/> do not have the same number of elements.</exception>
+        public BitArray Or(BitArray value) => Invoke<OrBinaryOp>(value);
+
+        /// <summary>
+        /// Performs the bitwise XOR operation between the elements of the current <see cref="BitArray"/> object and the
+        /// corresponding elements in the specified array. The current <see cref="BitArray"/> object will be modified to
+        /// store the result of the bitwise XOR operation.
+        /// </summary>
+        /// <param name="value">The array with which to perform the bitwise XOR operation.</param>
+        /// <returns>An array containing the result of the bitwise XOR operation, which is a reference to the current <see cref="BitArray"/> object.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="value"/> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="value"/> and the current <see cref="BitArray"/> do not have the same number of elements.</exception>
+        public BitArray Xor(BitArray value) => Invoke<XorBinaryOp>(value);
+
+        /// <summary>
+        /// Inverts all the bit values in the current <see cref="BitArray"/>, so that elements set to true are changed to false,
+        /// and elements set to false are changed to true.
+        /// </summary>
+        /// <returns>The current instance with inverted bit values.</returns>
+        public BitArray Not()
+        {
+            Invoke<NotBinaryOp>(this); // argument is ignored
+            ClearHighExtraBits(); // applying ~ to last Int32 may set extra bits we're trying to keep clear
+            return this;
+        }
+
+        /// <summary>Provides the implementation for <see cref="And"/>, <see cref="Or"/>, etc.</summary>
+        private BitArray Invoke<TBinaryOp>(BitArray value) where TBinaryOp : struct, IBinaryOp
```

Comment:
Definitely like the simplification here.

It would be nice if (long term, not this PR), we could use the more robust helpers from TensorPrimitives instead. Such helpers would be reusable for a few other locations across S.P.Corelib as well.

Reply:
Yeah, I contemplated building some of TensorPrimitives into corelib, but it wasn't super clean.